src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -248,6 +256,69 @@ public Location interiorCenter()
         return new JtsLocationConverter().backwardConvert(point.getCoordinate());
     }
 
+    /**
+     * @param threshold
+     *            {@link Angle} threshold that decides whether a {@link Heading} difference between
+     *            segments should be counted towards heading change count or not
+     * @return true if this {@link Polygon} has an approximately triangular shape while ignoring
+     *         {@link Heading} differences between inner segments that are below given threshold. A
+     *         {@link Polygon} might have more than 3 shape points with slight heading differences.
+     */
+    public boolean isApproximatelyTriangle(final Angle threshold)
```

Comment:
This could possibly be updated to handle more than triangles and other shapes by giving the number of sides as a parameter.